tubes/protocol.py

```diff
@@ -274,6 +278,22 @@ def connectionLost(self, reason):
             self._drain.fount.stopFlow()
 
 
+    def readConnectionLost(self):
+        """
+        An end-of-file was received.
+        """
+        self._fount.drain.flowStopped(Failure(ConnectionDone()))
+        self._fount.drain = None
```

Comment:
Ditto for later -- can we set it to something that's not None? There's this guy who wrote about why it sucks to get exceptions like "None object doesn't have a method "flowStopped"" is not great. Is there a DeadDrain/DeadFount object that will give more sensible exceptions?

Or just deleting the .drain attribute from self._fount altogether?

I mean, we're already playing fast and loose with Demeter anyway...

(Ditto for writeConnectionLost)

Reply:
This is great feedback, but it goes outside the scope of this branch somewhat.  Right now a `None` drain is the way that "not draining anywhere" is signaled, and this is a public attribute.

The interface doesn't call this out though, and perhaps a future change could address it, just not here :).